vignettes/ebola.Rmd

```diff
@@ -0,0 +1,615 @@
+* `stime_upr`: The upper bound on the secondary observation time i.e. `stime_lwr + 1`
+* `delay_daily`: Is given by `stime_daily - ptime_daily`
+* `delay_lwr`: Is given by `stime_lwr - ptime_upr` i.e. the minimum possible delay
+* `delay_upr`: Is given by `stime_upr - ptime_lwr` i.e. the maximum possible delay
+* `obs_at`: The maximum value of `stime`
+
+## Model fitting
+
+Now we are ready to fit lognormal delay distributions to the data!
+First, we start with a simple model that assumes the same delay distribution for all cases.
+
+```{r}
+subsample <- 0.2
+```
+
+For the time being, we filter the data to only complete cases (i.e. rows of the data which have no missing values^[It should be a relatively simple extension to allow for missing data in the model.]).
+As well, to speed up computation, we take a random `r 100 * subsample`% subsample of the complete data.
+
+```{r}
+set.seed(1)
+n <- nrow(obs_cens)
+obs_cens <- obs_cens[complete.cases(obs_cens)]
+n_complete <- nrow(obs_cens)
+n_complete / n
+obs_cens <- obs_cens[sample(seq_len(.N), n_complete * subsample, replace = FALSE)]
+```
+
+To prepare the data for use with the latent individual model, we use the function `as_latent_individual`:
+
+```{r}
+obs_prep <- as_latent_individual(obs_cens)
+```
+
+Now we are ready to fit the latent individual model.
+
+### Intercept only model
+
+We set `formula` to `brms::bf(mu ~ 1, sigma ~ 1)` to place an intercept only model on the `mu` and `sigma` parameters of the lognormal distribution.
+This distribution is specified using `family = brms::lognormal()`.
+We use the Laplace approximation via `algorithm = "laplace"` (see the approximate inference vignette) for fast model fitting.
+
+```{r}
+fit <- epidist(
+  data = obs_prep,
+  formula = brms::bf(mu ~ 1, sigma ~ 1),
+  family = brms::lognormal(),
+  algorithm = "laplace",
+  refresh = 0,
+  silent = 2,
+  seed = 1
+)
+```
+
+The `fit` object is a [`brmsfit`](https://paulbuerkner.com/brms/reference/brmsfit-class.html) object, and has the associated range of methods.
+See `methods(class = "brmsfit")` for more details.
+For example, we may use `summary` to view information about the fitted model, including posterior estimates for the regression coefficients:
+
+```{r}
+summary(fit)
+```
+
+### Sex-stratified model
+
+To fit a model which varies `mu` and `sigma` by sex we alter the `formula` specification to include fixed effects for sex as follows:
+
+```{r}
+fit_sex <- epidist(
+  data = obs_prep,
+  formula = brms::bf(mu ~ 1 + sex, sigma ~ 1 + sex),
+  family = brms::lognormal(),
+  algorithm = "laplace",
+  refresh = 0,
+  silent = 2,
+  seed = 1
+)
+```
+
+A summary of the model shows that males tend to have longer delays (the posterior mean of `sexM` is `r signif(summary(fit_sex)$fixed[3, "Estimate"], 3)`) and greater delay variation (the posterior mean of `sigma_sexM` is `r signif(summary(fit_sex)$fixed[4, "Estimate"], 3)`):
```

Comment:
```suggestion
A summary of the model shows that males tend to have longer delays (the posterior mean of `sexM` is `r signif(summary(fit_sex)$fixed[3, "Estimate"], 1)`) and greater delay variation (the posterior mean of `sigma_sexM` is `r signif(summary(fit_sex)$fixed[4, "Estimate"], 1)`):
```

Reply:
I think this will match the format of the Regression Coefficients table printed to the console

Reply:
I think we have to assume people have some frequentist instincts and so we may want to say something to help people interpret this correctly - e.g. point out that the posteriors are still highly overlapping here, and so the difference exists but may or may not be meaningful depending on the context

Reply:
[b0b2854](https://github.com/epinowcast/epidist/pull/52/commits/b0b2854511342d36c9224419dacbc6cd3a5f74a1)